detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/SwallowedException.kt

```
@@ -77,8 +77,14 @@ class SwallowedException(config: Config = Config.empty) : Rule(config) {
         "The caught exception is swallowed. The original exception could be lost.",
         Debt.TWENTY_MINS)
 
-    private val ignoredExceptionTypes = valueOrDefaultCommaSeparated(IGNORED_EXCEPTION_TYPES, emptyList())
-        .map { it.removePrefix("*").removeSuffix("*") }
+    private val defaultIgnoredExceptions = listOf(
+            DefaultIgnoredExceptionTypes.INTERRUPTED_EXCEPTION,
+            DefaultIgnoredExceptionTypes.NUMBER_FORMAT_EXCEPTION,
+            DefaultIgnoredExceptionTypes.PARSE_EXCEPTION,
+            DefaultIgnoredExceptionTypes.MALFORMED_URL_EXCEPTION
+    ).map { it.toString() }
+    private val ignoredExceptionTypes = valueOrDefaultCommaSeparated(IGNORED_EXCEPTION_TYPES, defaultIgnoredExceptions)
+            .map { it.removePrefix("*").removeSuffix("*") }
```

I think that a hardcoded list of strings is cleaner here. We are not going to use the enum for anything else so just the list is more simple.

---

The configuration option and not the code itself should be adapted. You might ask why?
If a detekt user doesn't want to have `NumberFormatException` as an ignored exception, the behavior can't be changed. Hence, the following lines should be touched. The code can stay as it is.

https://github.com/detekt/detekt/blob/c8890fa5133154eec4840260fd61281d8d43e919/detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/exceptions/SwallowedException.kt#L66-L70